src/panels/config/energy/dialogs/dialog-energy-grid-flow-settings.ts

```diff
@@ -67,18 +69,31 @@ export class DialogEnergyGridFlowSettings
         ]
       ? "statistic"
       : "no-costs";
-    this._pickedDisplayUnit = getDisplayUnit(
-      this.hass,
+
+    const initialSourceId =
       this._source[
         this._params.direction === "from"
           ? "stat_energy_from"
           : "stat_energy_to"
-      ],
+      ];
+
+    this._pickedDisplayUnit = getDisplayUnit(
+      this.hass,
+      initialSourceId,
       params.metadata
     );
     this._energy_units = (
       await getSensorDeviceClassConvertibleUnits(this.hass, "energy")
     ).units;
+
+    this._excludeList = [
+      ...(this._params.current_preference?.flow_from?.map(
+        (entry) => entry.stat_energy_from
+      ) || []),
+      ...(this._params.current_preference?.flow_to?.map(
+        (entry) => entry.stat_energy_to
+      ) || []),
+    ].filter((id) => id !== initialSourceId);
```

Comment:
Should we exclude any other non-grid entities as well?

Reply:
Thank you for working on this.

I am not sure that your second exclusion (prevent adding to supply and consumer) is correct.

A battery could be a consumer when charging but a supplier e.g. during the night.
I suspect there are other use cases also.

Reply:
> A battery could be a consumer when charging but a supplier e.g. during the night.
> I suspect there are other use cases where users might want to have an entity in both.

Wouldn't that case need to be two separate entities, like battery_charge_energy and battery_discharge_energy?

If you add the same _entity_ to both sections I would think they just cancel each other out 100% of the time.

Reply:
That's true.
Hmmm. Yes, can't think of a contradicting case then. I recall once abusing the two to be able to compare values against each other. But that was for convenience, not necessity. And it was not the originally intended use :-D

Reply:
> Should we exclude any other non-grid entities as well?

I guess that would make sense. Any entity should not appear twice anywhere on the Energy dashboard (otherwise, things will be wrong for sure).

Reply:
Also the "Individual Devices" section allows the inclusion of duplicate devices as well.